superset-frontend/plugins/plugin-chart-pivot-table/src/plugin/transformProps.ts

```diff
@@ -176,5 +177,6 @@ export default function transformProps(chartProps: ChartProps<QueryFormData>) {
     onContextMenu,
     timeGrainSqla,
     allowRenderHtml,
+    dataSelectionMode,
```

Comment:
Here is the place to pass the appropriate data selection mode value for the given viz plugin.
```suggestion
    dataSelectionMode: canExportData ? 'auto' : 'none'
```